table/builder.go

```diff
@@ -273,10 +298,40 @@ func (b *Builder) EstimateSize() int {
 // Finish finishes the table by appending the index.
 func (b *Builder) Finish() error {
 	b.finishBlock() // This will never start a new block.
-	b.buf = append(b.buf, u32SliceToBytes(b.blockEndOffsets)...)
-	b.buf = append(b.buf, b.baseKeysBuf...)
-	b.buf = append(b.buf, u32SliceToBytes(b.baseKeysEndOffs)...)
+	if err := b.w.Finish(); err != nil {
+		return err
+	}
+	idxFile, err := y.OpenTruncFile(b.idxFileName, false)
+	if err != nil {
+		return err
+	}
+	b.w.Reset(idxFile)
+
+	var encodeBuf [8]byte
+
+	// Don't compress the global ts, because it may be updated during ingest.
+	ts := uint64(math.MaxUint64)
+	if b.isExternal {
+		// External builder doesn't append ts to the keys, the output sst should has a non-MaxUint64 global ts.
+		ts = math.MaxUint64 - 1
+	}
+	binary.BigEndian.PutUint64(encodeBuf[:], ts)
+	if err := b.w.Append(encodeBuf[:]); err != nil {
+		return err
+	}
+
+	binary.BigEndian.PutUint64(encodeBuf[:], uint64(b.writtenLen))
+	b.buf = append(b.buf, encodeBuf[:]...)
+
+	b.buf = append(b.buf, u32ToBytes(uint32(len(b.smallest)))...)
```

Comment:
I think it's time we add a field flag for each index field, so later adding a new field would be compatible.